### ISSUE TITLE
Using passport without login

### ISSUE THREAD

**Issue by jolacaleb** [meanjs/mean#854]
Hello! I'm new to meanjs and developing with this amazing framework. But now I'm stuck with a problem, so I ask question here with hope somebody can answer my problem.

I'm having a hard time to get a information from facebook app. I set passport information, app_id, and app_secret properly for I can properly login to my website as manual.

But what I want to do is that I only want to retrieve the info from facebook, not automatically login because there are more informations user should give.

I'm using meanJs 0.3.x.
```html
<a href="/auth/facebook/">
```
works properly.
I tried to retrieve the data only, what I have done is,

**users.authentication.server.controller.js**
```javascript
/**
 * OAuth callback
 */
exports.oauthCallback = function(strategy) {
	return function(req, res, next) {
		passport.authenticate(strategy, function(err, user, redirectURL) {
			if (err || !user) {
				return res.redirect('/#!/signin');
			}

			res.json(user);
			/**
			req.login(user, function(err) {
				if (err) {
					return res.redirect('/#!/signin');
				}

				return res.redirect(redirectURL || '/');
			});
			 */
		})(req, res, next);
	};
};
```
I commented out <code>req.login</code>function not to login.

and client side...

**signup.client.controller.js**
```javascrcipt
		$scope.signupWithFacebook = function() {
			$http.get('/auth/facebook/').success(function(result) {
				$scope.userData = result;
				console.log(result);
			});
		};
```
There is error at the client side which is
<code>XMLHttpRequest cannot load https://www.facebook.com/dialog/oauth?..... No 'Access-Control-Allow-Origin' header is present on the requested resource. Origin 'http://localhost:3000' is therefore not allowed access.</code>

Is there any way to retrieve facebook user's information without login? Any help can be appreciated.

**Comment by codydaig** [comment]
@jolacaleb Are you referring to when a user sign's up or logs in?

**Comment by jolacaleb** [comment]
@codydaig When user signs up. At this time, when user clicks 'Signup with Facebook', then the user automatically logs in, even though our website needs more additional information to register.

The mechanism I'm trying to implement is, if a user clicks Signup with Facebook button, basic information automatically fills the form, not automatically logging in.

**Comment by codydaig** [comment]
@jolacaleb Sorry for the delay. Must have accidentally cleared the notification without responding. 

Personally, I've never tried too do that. If I had to come up with a solution for that though, I would probably make the signup process a two step process for all users (whether they are a local signup or auth signup like facebook, github, etc...) so the initial signup would just create their account. Then they would be logged in, but redirected to a page to gather more information from them. At that point, you can auto populate fields if they have you access to them in their auth service provider. Then don't attatch the 'user' role to the user object until that form is complete, so anything they must be logged in and have completed the form for, they won't be able to see until the form is completed.

**Comment by jolacaleb** [comment]
@codydaig Oh I solved this by not using passport.js. I looked how it works from the docs, I finally concluded that with using passport.js, I can only login, and only after login I can get the information. I'm not that good in English so I might be wrong.
 Anyway I solved this with using Facebook SDK provided by developer.facebook.com. If the user authorized his/her facebook account to allow our app, my site can get informations about him/her without logging in.
 I should commented here that it was solved, but I completely forgot I left a question here. Now i'm gonna close this. Thanks for helpful comment!
